ctt-frontend/src/app/services/aws-category.service.spec.ts

```diff
@@ -1,16 +1,217 @@
-// import { TestBed } from '@angular/core/testing';
+import { TestBed } from '@angular/core/testing';
+import * as TypeMoq from 'typemoq';
+import { APIService } from './API.service';
 
-// import { AwsCategoryService } from './aws-category.service';
+import { AwsCategoryService } from './aws-category.service';
+import {
+  CreateCategoryInput as APICreateInput,
+  UpdateCategoryMutation,
+} from './API.service';
+import * as DummyData from './test-data/aws-category-service-data';
 
-// describe('AwsCategoryService', () => {
-//   let service: AwsCategoryService;
+describe('AwsCategoryService', () => {
+  let service: AwsCategoryService;
 
-//   beforeEach(() => {
-//     TestBed.configureTestingModule({});
-//     service = TestBed.inject(AwsCategoryService);
-//   });
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(AwsCategoryService);
+  });
 
-//   it('should be created', () => {
-//     expect(service).toBeTruthy();
-//   });
-// });
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('create', () => {
+    it('should accept a CreateCategoryInput', async () => {
+      const apiMock = TypeMoq.Mock.ofType(APIService);
+      apiMock
+        .setup((x) =>
+          x.CreateCategory(TypeMoq.It.is((y) => (y as APICreateInput) !== null))
+        )
+        .returns((y) => Promise.resolve(DummyData.createMutation));
+      const sut = new AwsCategoryService(apiMock.object);
+
+      await expectAsync(sut.create(DummyData.categoryInput)).toBeResolvedTo(
+        DummyData.categoryValue
+      );
+      expect(
```

Comment:
Also the `expect(...)` wrapping around the `apiMock.verify(...)` call is redundant and can be removed. `verify` throws when no call is found and so the test fails.